scikeras/utils/transformers.py

```diff
@@ -0,0 +1,196 @@
+        self.loss = loss
+
+    def fit(self, X: np.ndarray) -> "ClassifierLabelEncoder":
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        target_type = type_of_target(y)
+        encoders = {
+            "binary": make_pipeline(
+                Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
+            ),
+            "multiclass": make_pipeline(
+                Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
+            ),
+            "multiclass-multioutput": FunctionTransformer(),
+            "multilabel-indicator": FunctionTransformer(),
+        }
+        if is_categorical_crossentropy(self.loss):
+            encoders["multiclass"] = make_pipeline(
+                Ensure2DTransformer(), OneHotEncoder(sparse=False, dtype=np.float32),
+            )
+        if target_type not in encoders:
+            raise ValueError(
+                f"Unknown label type: {target_type}."
+                "\n\nTo implement support, subclass KerasClassifier and override"
+                " `target_transformer` with a transformer that supports this"
+                " label type."
+            )
+        self.final_encoder_ = encoders[target_type].fit(y)
+
+        if (
+            target_type == "multilabel-indicator"
+            and
+            np.all(np.sum(y, axis=1)==1)
+        ):
+            target_type = "multiclass-onehot"
+
+        if target_type in ["binary", "multiclass"]:
+            self.classes_ = self.final_encoder_[1].categories_[0]
+            self.n_classes_ = self.classes_.size
+        elif target_type == "multiclass-onehot":  # one-hot encoded multiclass
+            self.classes_ = np.arange(0, y.shape[1])
+            self.n_classes_ = y.shape[1]
+        else:
+            self.classes_ = None
+            self.n_classes_ = None
+        self.n_outputs_ = 1
+        self.model_n_outputs_ = 1
+        self.y_dtype_ = y.dtype
+        self._target_type = target_type
+        return self
+
+    def transform(self, X: np.ndarray) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        # no need to validate n_outputs_ or model_n_outputs_, those are hardcoded
+        # self.classes_ and self.n_classes_ are validated by the transformers themselves
+        return self.final_encoder_.transform(y)
+
+    def inverse_transform(
+        self, X: np.ndarray, return_proba: bool = False
+    ) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        if self._target_type == "binary":
+            # array([0.9, 0.1], [.2, .8]) -> array(['yes', 'no'])
+            if self.n_classes_ == 1:
+                # special case: single input label for sigmoid output
+                # may give more predicted classes than inputs for
+                # small sample sizes!
+                # don't even bother inverse transforming, just fill.
```

Comment:
I think the issue here is that you can get  a prediction `[0, 0, 1]` even if your input targets was `[0, 0, 0]` which would throw an error from the transformer. I now want to double check what dataset/model I saw this with, because I don't remember, and there's always a chance it was set up wrong.

Reply:
I was able to get rid of this. One of two things is happened/is happening:
(1) I set up the model where I first observed this wrong, or it was due to some other bug in the wrapper code. I think this is the most likely.
(2) It's an intermittent problem stemming from the probabilistic nature of the nonlinearity. I kind doubt this but don't discard it completely.

Reply:
Anyway since the branch is deleted (at least for now) I'll resolve this.